packages/optimizer/lib/transformers/OptimizeImages.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const {hasAttribute, nextNode, firstChildByTag} = require('../NodeUtils');
+const {skipNodeAndChildren} = require('../HtmlDomHelper');
+const {isValidImageSrcURL} = require('../URLUtils');
+
+// Don't generate srcsets for images larger than MAX_IMG_SIZE
+const MAX_IMG_SIZE = 1500;
+
+// Don't generate srcset's for images with width smaller than MIN_WIDTH_TO_ADD_SRCSET_IN_RESPONSIVE_LAYOUT
+// this avoids generating srcsets for images with a responsive layout where width/height define the aspect ration.
+const MIN_WIDTH_TO_ADD_SRCSET_IN_RESPONSIVE_LAYOUT = 100;
+
+// All legimate srcset widths.
+const SRCSET_WIDTH = [39, 56, 82, 100, 150, 300, 500, 750, 1000, 1500, 2000, 2500];
+
+// The maximum number of values. We'll take the initial image width and generate more width values by
```

Comment:
does "values" in this context mean "maximum number of [`srcset` sources]"?

Reply:
yes, rephrased